python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,307 @@
+                    be sent between nodes unencrypted.
+                    """,
+                    RuntimeWarning,
+                )
+                return
+            raise RuntimeError(
+                f"""
+                This cluster has TLS encryption enabled;
+                however, {name} does not support
+                data encryption in transit. To override
+                this configuration and use {name}
+                anyway, you may set '{self.ssl_conf}'
+                to 'true' in the Spark configuration. Please note this
+                will cause model parameters and possibly training
+                data to be sent between nodes unencrypted.
+                """
+            )
+
+
+class TorchDistributor(Distributor):
+    """
+    A class to support distributed training on PyTorch and PyTorch Lightning using PySpark.
+
+    .. versionadded:: 3.4.0
+
+    Examples
+    --------
+
+    Run PyTorch Training locally on GPU (using a PyTorch native function)
+
+    >>> def train(learning_rate):
+    ...     import torch.distributed
+    ...     torch.distributed.init_process_group(backend="nccl")
+    ...     # ...
+    ...     torch.destroy_process_group()
+    ...     return model # or anything else
+    >>> distributor = TorchDistributor(
+    ...     framework="pytorch",
+    ...     num_processes=2,
+    ...     local_mode=True,
+    ...     use_gpu=True)
+    >>> model = distributor.run(train, 1e-3)
+
+    Run PyTorch Training on GPU (using a file with PyTorch code)
+
+    >>> distributor = TorchDistributor(
+    ...     framework="pytorch",
+    ...     num_processes=2,
+    ...     local_mode=False,
+    ...     use_gpu=True)
+    >>> distributor.run("/path/to/train.py", *args)
+
+    Run PyTorch Lightning Training on GPU
+
+    >>> num_proc = 2
+    >>> def train():
+    ...     from pytorch_lightning import Trainer
+    ...     # ...
+    ...     # required to set devices = 1 and num_nodes == num_processes
+    ...     trainer = Trainer(accelerator="gpu", devices=1, num_nodes=num_proc, strategy="ddp")
+    ...     trainer.fit()
+    ...     # ...
+    ...     return trainer
+    >>> distributor = TorchDistributor(
+    ...     framework="pytorch-lightning",
+    ...     num_processes=num_proc,
+    ...     local_mode=True,
+    ...     use_gpu=True)
+    >>> trainer = distributor.run(train)
+    """
+
+    # TODO(SPARK-41915): Remove need for setting frameworks in a future PR.
+    available_frameworks = ["pytorch", "pytorch-lightning"]
+
+    def __init__(
+        self,
+        framework: str,
+        num_processes: int = 1,
+        local_mode: bool = True,
+        use_gpu: bool = True,
+    ):
+        """Initializes the distributor.
+
+        Parameters
+        ----------
+        framework : str
+            A string indicating whether or not we are using PyTorch or PyTorch
+            Lightning. This could either be the string “pytorch” or ”pytorch-lightning”.
```

Comment:
Seems like you're using special character backquotes (from Mac by default). It would have to be `"` instead of `“`

Reply:
Thanks for catching that!